lessons/Lesson-4-Ruby-Part-2.md

```diff
@@ -10,71 +11,111 @@ stringArray.forEach((str) => {
 })
 ```
 
-```
+```ruby
 # Next, in Ruby, with a block
 
 string_array = ["first", "second", "third"]
 
-string_array.each do | str | # you could also have { | str| puts str }
+string_array.each do |str|
   puts str
 end
 ```
-The appearance is a little different.  You surround the value that is to be passed on each iteration with vertical bars.  Some blocks are passed multiple values, as in the following:
+
+If you only need one line within the block, then you can use `{ ... }` on the same line instead of `do ... end`:
+
+```ruby
+string_array.each { |str| puts str }
+```
+
+The appearance is a little different.  You surround the value that is to be passed on each iteration with vertical bars.
+Additionally, if you wish to refer to block arguments by their position instead of having to name them:
+
+```ruby
+string_array.each { puts _1 } # `_1` refers to the first block argument, `_2` to the second, etc.
```

Comment:
This has been around since Ruby 2.7 (2019), and I've seen it around [often enough](https://github.com/search?type=code&q=language%3Aruby+path%3A**.rb+%2F%5C%7B+.*%5Cb_%5B1-9%5D%5Cb%2F) that I think it's worth going over here, even if the later material doesn't use it.